value/sys.go

```diff
@@ -39,53 +43,129 @@ To convert a time vector to a seconds value:
 To print seconds in Unix date format:
   'T' text sys 'sec'`
 
+func vecText(v Vector) string {
+	s := fmt.Sprint(v) // will print as "(text)"
+	return s[1 : len(s)-1]
+}
+
 // sys implements the variegated "sys" unary operator.
 func sys(c Context, v Value) Value {
-	conf := c.Config()
 	vv := v.(Vector)
-	if !allChars(vv) {
-		Errorf("sys %s not defined", v)
+
+	if allChars(vv) { // single argument
+		verb := vecText(vv)
+		if fn, ok := sys1[verb]; ok {
+			return fn(c)
+		}
+		if fn, ok := sys2[verb]; ok {
+			return fn(c, []Value{})
+		}
+		Errorf("sys %q not defined", verb)
+	}
+
+	if v1, ok := vv[0].(Vector); ok && allChars(v1) { // multiple arguments, verb first
+		verb := vecText(v1)
+		if fn, ok := sys2[verb]; ok {
+			return fn(c, vv[1:])
+		}
+		if _, ok := sys1[verb]; ok {
+			Errorf("sys %q takes no arguments", verb)
+		}
+		Errorf("sys %q not defined", verb)
 	}
-	arg := fmt.Sprint(vv) // Will print as "(argument)"
-	switch arg[1 : len(arg)-1] {
-	case "help":
-		fmt.Fprint(conf.Output(), sysHelp)
+
+	Errorf("sys requires string argument")
+	panic("unreachable")
+}
+
+var sys1 = map[string]func(c Context) Value{
+	"help": func(c Context) Value {
```

Comment:
conf := c.Config() will make the code (and the changes) notably shorter.

Reply:
Done, although I think it ends up longer, because `conf *config.Config; conf` is longer than `c Context; c.Config()`.